components/net/tests/local_directory_listing.rs

```diff
@@ -0,0 +1,740 @@
+        YearType::CurrentYear,
+        &now_year.as_str(),
+        "23",
+        "17",
+        "00",
+    );
+    write_expected_file_row(
+        &mut expected,
+        "java-log",
+        FileType::File,
+        "86.06",
+        BinarySizeUnit::Kibibytes,
+        format!("{now_year}-{now_month_iso}-{not_today_iso}T06:30:28.734").as_str(),
+        DateType::NotToday,
+        &not_today_display,
+        &not_today_suffix,
+        &now_month_display,
+        YearType::CurrentYear,
+        &now_year.as_str(),
+        "06",
+        "30",
+        "28",
+    );
+    write_expected_file_row(
+        &mut expected,
+        "latest",
+        FileType::Symlink,
+        "19",
+        BinarySizeUnit::Bytes,
+        format!("{now_year}-{now_month_iso}-{now_day_iso}T23:57:58.120").as_str(),
+        DateType::Today,
+        &now_day_display,
+        &today_suffix,
+        &now_month_display,
+        YearType::CurrentYear,
+        &now_year.as_str(),
+        "23",
+        "57",
+        "58",
+    );
+    write_expected_file_row(
+        &mut expected,
+        "rust-log",
+        FileType::File,
+        "13",
+        BinarySizeUnit::Bytes,
+        format!("2023-{now_month_iso}-{now_day_iso}T22:31:22.333").as_str(),
+        DateType::NotToday,
+        &now_day_display,
+        &today_suffix,
+        &now_month_display,
+        YearType::NotThisYear,
+        "2023",
+        "22",
+        "31",
+        "22",
+    );
+    write_expected_foooter_and_end(&mut expected);
+    let result = build_html_directory_listing(path, has_parent, items);
+    assert_eq!(result, expected);
+}
+
+fn write_expected_start_and_header(expected: &mut String, directory_label: &str, has_parent: bool) {
+    expected.push_str(
+        "<!DOCTYPE html>\
+<html lang=\"en\">\
+<head><title>Directory listing: ",
+    );
+    expected.push_str(directory_label);
+    expected.push_str("</title><style>");
+    expected.push_str(read_string(Resource::DirectoryListingCSS).as_str());
+    expected.push_str("</style></head><body>");
+    expected.push_str("<header><h1>Index of <span class=\"path\">");
+    expected.push_str(directory_label);
+    expected.push_str(
+        "</span>\
+        </h1></header>",
+    );
+    expected.push_str("<div class=\"directory_info\">");
+    if has_parent {
+        expected.push_str(
+            "<div class=\"parent_link\"><a href=\"../\">Up to parent directory</a></div>",
+        );
+    }
+    expected.push_str("<div class=\"listing\">");
+}
+
+fn write_expected_file_row(
+    expected: &mut String,
+    filename: &str,
+    is_symlink: FileType,
+    filesize: &str,
+    size_unit: BinarySizeUnit,
+    iso_datetime: &str,
+    day_is_current: DateType,
+    now_day_display: &str,
+    today_suffix: &str,
+    now_month_display: &str,
+    year_is_current: YearType,
+    now_year_display: &str,
+    now_hour_display: &str,
+    now_minute_display: &str,
+    now_second_display: &str,
+) {
```

Comment:
This function has too many arguments. Send the date as a single string, or better yet, pass in a SystemTime that is formatted here.